test/index.ts

```diff
@@ -1782,55 +1773,78 @@ describe('BigQuery', () => {
       });
     });
 
-    it('should call job#getQueryResults', done => {
-      const fakeJob = {
-        getQueryResults: (options: {}, callback: Function) => {
-          callback(null, FAKE_ROWS, FAKE_RESPONSE);
+    it('should call table#getRows', done => {
+      const TABLE_ID = 'bq_table';
+      const DATASET_ID = 'bq_dataset';
+      const metadata = {
+        configuration: {
+          query: {
+            destinationTable: {
+              datasetId: DATASET_ID,
+              tableId: TABLE_ID,
+            },
+          },
         },
       };
 
+      const fakeJob = new EventEmitter();
+      // tslint:disable-next-line: no-any
+      (fakeJob as any).metadata = metadata;
+
       bq.createQueryJob = (query: {}, callback: Function) => {
         callback(null, fakeJob, FAKE_RESPONSE);
       };
 
-      bq.query(QUERY_STRING, (err: Error, rows: {}, resp: {}) => {
+      const finalCallback = (err: Error | null, rows: {}, resp: {}) => {
         assert.ifError(err);
         assert.strictEqual(rows, FAKE_ROWS);
         assert.strictEqual(resp, FAKE_RESPONSE);
         done();
-      });
-    });
+      };
 
-    it('should assign Job on the options', done => {
-      const fakeJob = {
-        getQueryResults: (options: {}, callback: Function) => {
+      const fakeTable = {
+        getRows(options: {}, cb: Function) {
           assert.deepStrictEqual(options, {job: fakeJob});
-          done();
+          assert.strictEqual(cb, finalCallback);
```

Comment:
We don't really need to simulate executing the callback in this case, since we just want to know that the user's callback was passed to `table.getRows`. In other tests, we just use the `done` function as the simulated user callback, so that it just ends the test:

```js

const fakeTable = {
  getRows(options: {}, cb: Function) {
    assert.deepStrictEqual(options, {job: fakeJob});
    cb(); // done()
  }
};

// ...

bq.query(QUERY_STRING, done);
```

Either way works, if you still want to break out and name the user callback. I would just leave it at the strict comparison of cb === finalCallback in that case, and then just end the test right after.